tests/TestCase/Database/QueryTest.php

```diff
@@ -1709,18 +1709,17 @@ public function testWhereInArrayEmpty()
         $query = new Query($this->connection);
         $query->select(['id'])
             ->from('articles')
-            ->whereInList('id', [], ['allowEmpty' => true])
-            ->execute();
-        $sql = $query->sql();
-
-        $result = $query->execute();
-        $this->assertFalse($result->fetch('assoc'));
+            ->whereInList('id', [], ['allowEmpty' => true]);
 
         $this->assertQuotedQuery(
             'SELECT <id> FROM <articles> WHERE 1=0',
-            $sql,
+            $query->sql(),
             !$this->autoQuote
         );
+
+        $statement = $query->execute();
+        $this->assertFalse($statement->fetch('assoc'));
+        $statement->closeCursor();
```

Comment:
This closeCursor() is probably not necessary because fetch() returns false. The issue was the double execute() above.